plugins/modules/cloud/misc/terraform.py

```diff
@@ -80,9 +80,22 @@
     aliases: [ 'variables_file' ]
   variables:
     description:
-      - A group of key-values to override template variables or those in
-        variables files.
+      - A group of key-values pairs to override template variables or those in variables files.
+      - Support complex variable structures (Lists, Dictionaries and Boolean) to reflect terraform variable syntax when C(complex_vars=true).
+      - Ansible integers or floats are mapped to terraform numbers.
+      - Ansible strings are mapped to terraform strings.
+      - Ansible dictionaries are mapped to terraform objects.
+      - Ansible lists are mapped to terraform lists.
+      - Ansible booleans are mapped to terraform booleans.
+      - "B(Note) passwords passed as variables will be visible in the log output. Make sure to use I(no_log=true) in production!"
     type: dict
+  complex_vars:
+    description:
+      - Enable/disable capability to handle complex variable structures for C(terraform).
+      - If C(true) the C(variables) would accept I(Booleans), I(Objects) and I(Lists) to be passed to C(terraform).
```

Comment:
```suggestion
      - If C(true) the I(variables) also accepts dictionaries and lists to be passed to C(terraform).
        Strings that are passed are correctly quoted.
```

Reply:
The old code would not handle native Ansible Booleans, for example, if a variable `mybool` is of type bool, then `mybool: true` would translate to in the following code as: 

```
            variables_args.extend([
                '-var',
                '{0}={1}'.format(k, v)
            ])```

as `'-var mybool=True'` which is not acceptable to `terraform`. This is not a total loss, the way the old code worked , we can still pass a string, rather than a bool. For example, `mybool: 'true'`, would still correctly transform to `'-var mybool=true'`

That's the reason why I had the Boolean there.... do we need to explain that further? or do you think that confuses readers? I'll take your lead on this one :).

Reply:
Makes sense! I've upated my suggestion accordingly.